concordium-std/src/types.rs

```diff
@@ -1,11 +1,100 @@
-use crate::num::NonZeroU32;
+use crate::{num::NonZeroU32, HasContractStateEntry, HasContractStateLL};
+use std::{cell::RefCell, marker::PhantomData, rc::Rc};
```

Comment:
This part generally has to work with and without `std`. 

We re-export some definitions already in `lib.rs.` If these are not please add them there. But I hope we can remove the Rc.